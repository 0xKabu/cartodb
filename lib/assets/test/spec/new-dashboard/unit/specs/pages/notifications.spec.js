import {
  mount,
  createLocalVue
} from '@vue/test-utils';
import NotificationsPage from 'new-dashboard/pages/Notifications';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);

let store;
const $t = key => key;

const notificationsMock = [{
  html_body: '<p>Cupcake Ipsum dolor gingerbread chocolate. <em>Pudding</em> wafer ice cream. Powder ice cream carrot cake <strong>liquorice</strong> cookie oat cake.</p>',
  icon: 'alert',
  id: '288dfe6e-1a9d-4157-bd36-41cd8459af62',
  read_at: null,
  received_at: '2018-11-20T15:28:21.792Z'
},
{
  html_body: '<p>Holi parte 2</p>',
  icon: 'alert',
  id: '091a04a7-28d6-4fcf-bd8a-b1c34842bdb4',
  read_at: null,
  received_at: '2018-11-19T17:11:42.285Z'
}
];

describe('NotificationsPage.vue', () => {
  let notificationsPageWrapper;
  describe('when there are notifications', () => {
    beforeEach(() => {
      store = new Vuex.Store({
        state: {
          user: {
            organizationNotifications: [notificationsMock]
          }
        }
      });
      notificationsPageWrapper = mount(NotificationsPage, {
        store,
        localVue,
        mocks: {
          $t
        }
      });
    });
    it('should show the nofications list', () => {
      const notificationList = notificationsPageWrapper.find('.notifications-list');
      expect(notificationList.exists()).toBe(true);
    });

    it('should hide the no notifications message', () => {
      const notificationList = notificationsPageWrapper.find('.empty-state');
      expect(notificationList.exists()).toBe(false);
    });
  });

  describe('when there are no notifications', () => {
    beforeEach(() => {
      store = new Vuex.Store({
        state: {
          user: {
            organizationNotifications: []
          }
        }
      });
      notificationsPageWrapper = mount(NotificationsPage, {
        store,
        localVue,
        mocks: {
          $t
        }
      });
    });
    it('should hide the nofications list', () => {
      const notificationList = notificationsPageWrapper.find('.notifications-list');
      expect(notificationList.exists()).toBe(false);
    });

    it('should show the no notifications message', () => {
      const notificationList = notificationsPageWrapper.find('.empty-state');
      expect(notificationList.exists()).toBe(true);
    });
  });
});
