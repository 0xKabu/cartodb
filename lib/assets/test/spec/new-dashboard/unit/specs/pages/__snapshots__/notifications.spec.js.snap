// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsPage.vue when notifications are being fetched should show a spinner 1`] = `
<section class="page">
  <div class="notifications-list-container container grid">
    <div class="full-width">
      <header class="head-section grid-cell">
        <div class="head-sectionContainer">
          <h2 class="head-sectionTitle is-txtGrey"><span class="icon icon--large"><img src="../assets/icons/section-title/envelope.svg"></span>
            Notifications
          </h2>
          <ul class="head-sectionActions">
            <li class="head-sectionActionsItem head-sectionDropdownContainer"></li>
            <!---->
          </ul>
        </div>
        <!---->
      </header>
      <ul class="notifications-list grid-cell  grid-cell--col9 grid-cell--col12--tablet"></ul>
      <!---->
      <div class="loading-state">
        <div class="loading-state-icon"><img svg-inline="" src="../../assets/icons/navbar/search/loading.svg" class="loading-state__svg"></div>
        <h6 class="loading-state-text text is-caption is-txtSoftGrey">fake_loading_text</h6>
      </div>
    </div>
  </div>
</section>
`;
